tensilelite/Tensile/Components/SIA.py

```diff
@@ -170,13 +170,36 @@ def getLocalWriteMFMAEnd(writer, kernel, tensorParametersA, tensorParametersB):
     isLocalReadsOpt = False
     tmpLatencyLeft  = 0
     tmpNumMfmaForLR = 0
-    localReadsNotWaited = writer.states.numReadsPerIterB//kernel["InnerUnroll"] - writer.states.numReadsPerUnrollB
+    localReadsNotWaited = writer.states.numReadsPerIterB // kernel["InnerUnroll"] - writer.states.numReadsPerUnrollB
     if kernel["UnrollMajorLDSB"] and localReadsNotWaited > 0:
         isLocalReadsOpt = True
 
+    # Add space to avoid LR FIFO stall
+    # 24 quad-cycle for b128
+    # no stall happen for b64/b32/b16
+    lrStallLatencyBuffer = 24
+    localReadFIFO = []
+    def checkLocalReadFIFO(currentMFMA, blockWidth):
+        if blockWidth < 4:
+            return False
+        # The FIFO length is 16 so that each wave has 16/numWaves buffer.
+        numWaves = kernel["MIWaveGroup"][0] * kernel["MIWaveGroup"][1]
```

Comment:
Don't we need to consider LocalSplitU case?
Number of wave is MIWG0 * MIWG1 * LSU.

Reply:
good catch. fixed